Admin/Client/src/App.tsx

```diff
@@ -1,7 +1,57 @@
-import React from 'react';
-
-function App() {
-    return <div className='App'></div>;
-}
-
-export default App;
+import React, { useState } from 'react';
+import InitSideEffects from './sideEffects/initSideEffects';
+import LoginPage from './pages/loginPage';
+import Authentication from './authentication';
+
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route,
+    Link,

+    Redirect
+} from "react-router-dom";
+import MainPage from './pages/mainPage';
+
+//Change this to move from real to mock side effects
+const isMocking = true;
+InitSideEffects(isMocking);
+
+function App() {
+    const [isLoggedin, setLoginStatus] = useState(false);
+    const login = (username: string, password: string, onStart:()=>void, onFail:(error:string)=>void, onSuccess:()=>void) => {
+        onStart();
+        Authentication.AuthenticateAsync(username, password)
+            .then(token => {
+                setLoginStatus(true);
+            })
+            .catch(err => {
+                onFail(err);
+            })
+            .finally(() => {
+                onSuccess();

```

Comment:
the success runs also after failure?

Reply:
my bad - naming issue - it should be onFinish or something